xmind/tests/test_topic_element.py

```diff
@@ -0,0 +1,267 @@
+    TAG_CHILDREN,
+    TAG_SHEET,
+    ATTR_ID,
+    ATTR_HREF,
+)
+
+
+class TestTopicElement(base.Base):
+    """Test class for TopicElement class"""
+
+    def getLogger(self):
+        if not getattr(self, '_logger', None):
+            self._logger = logging.getLogger('TopicElement')
+        return self._logger
+
+    def setUp(self):
+        super(TestTopicElement, self).setUp()
+        self._workbook_mixin_element_init = self._init_patch_with_name(
+            '_mixin_init', 'xmind.core.topic.WorkbookMixinElement.__init__')
+        self._add_attribute = self._init_patch_with_name(
+            '_add_attribute', 'xmind.core.topic.TopicElement.addIdAttribute', return_value=True)
+
+    def _assert_init_methods(self):
+        self._workbook_mixin_element_init.assert_called_once_with(None, None)
+        self._add_attribute.assert_called_once_with(ATTR_ID)
+
+    def test_excessive_parameters(self):
+        _element = TopicElement()
+        self.assertEqual(TAG_TOPIC, _element.TAG_NAME)
+
+        _parameters = [
+            ('_get_title', 0),
+            ('_get_markerrefs', 0),
+            ('_get_position', 0),
+            ('_get_children', 0),
+            ('_set_hyperlink', 1),
+            ('getOwnerSheet', 0),
+            ('getTitle', 0),
+            ('setTitle', 1),
+            ('getMarkers', 0),
+            ('addMarker', 1),
+            ('setFolded', 0),
+            ('getPosition', 0),
+            ('removePosition', 0),
+            ('getType', 0),
+            ('getTopics', (1, False)),
+            ('getSubTopics', (1, False)),
+            ('getSubTopicByIndex', 2),
+            ('addSubTopic', (3, False)),
+            ('getIndex', 0),
+            ('getHyperlink', 0),
+            ('setFileHyperlink', 1),
+            ('setTopicHyperlink', 1),
+            ('setURLHyperlink', 1),
+            ('getNotes', 0),
+            ('_set_notes', 0),
+            ('setPlainNotes', 1),
+        ]
+
+        for pair in _parameters:
+            with self.subTest(pair=pair):
+                self._test_method_by_excessive_parameters(pair, _element)
+
+        self._assert_init_methods()
+
+    def test_init_has_no_node_has_no_owner_workbook(self):
+        _element = TopicElement()
+        self._assert_init_methods()
+
+    def test_init_has_no_node_has_owner_workbook(self):
+        _element = TopicElement(ownerWorkbook=5)
+        self._workbook_mixin_element_init.assert_called_once_with(None, 5)
+        self._add_attribute.assert_called_once_with(ATTR_ID)
+
+    def test_init_has_node_has_no_owner_workbook(self):
+        _element = TopicElement(3)
+        self._workbook_mixin_element_init.assert_called_once_with(3, None)
+        self._add_attribute.assert_called_once_with(ATTR_ID)
+
+    def test_init_has_node_has_owner_workbook(self):
+        _element = TopicElement(3, 5)
+        self._workbook_mixin_element_init.assert_called_once_with(3, 5)
+        self._add_attribute.assert_called_once_with(ATTR_ID)
```

Comment:
shoul we also call it with 3 parameters?

Reply:
Addressed